envoy/http/codec.h

```diff
@@ -291,6 +291,11 @@ class StreamCallbacks {
    * watermark to under its low watermark.
    */
   virtual void onBelowWriteBufferLowWatermark() PURE;
+
+  /**
+   * Fires when a stream is ended (half-closed).
+   */
+  virtual void onStreamEnd() PURE;
```

Comment:
what this is doing isn't clear to me.   You can have end stream for request path and response path?

Reply:
Yes, this is a per-stream end, ES in diagram https://httpwg.org/specs/rfc9113.html#StreamStates.